codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/GoIntegration.java

```diff
@@ -111,6 +113,46 @@ default void onShapeWriterUse(
         // pass
     }
 
+    /**
+     * Populates operation middleware stack.
+     *
+     * This method is used to populate middleware stack inside
+     * the operation function.
+     *
+     * @param settings Settings used to generate.
+     * @param model Model to generate from.
+     * @param symbolProvider Symbol provider used for codegen.
+     * @param writer Writer that will be used.
+     * @param serviceShape Service shape of the operation.
+     * @param operationShape Operation Shape for which middleware stack is populated.
+     * @param protocolGenerator Protocol Generator associated with protocol used by the client.
+     */
+    default void assembleMiddlewareStack(
```

Comment:
This should be part of the `RuntimeClientPlugin`, also it would be simpler to just delegate to a static function that takes the stack and adds what is needed.